File: Messages/Server/Checksum.py
from Messages.Message import Message

class ChecksumMessage(Message):
    def __init__(self, *args):
        super().__init__(*args)
        self.id = 1338

    def decode(self):
        pass

    def encode(self):
        checksum = self.storage.checksum.encode("utf8")
        self.stream.writeU8(len(checksum))
        self.stream.writeBytes(checksum)

    def process(self):
        pass

    def is_client_to_server(self):
        return False

File: Protocol/Checksum.ts
import ByteStream from "bytestream.ts";
import Message from "./Message";

export default class ChecksumMessage extends Message {
    checksum: string;

    constructor(stream?: ByteStream) {
        super(stream);
        this.id = 51;
    }

    encode(): void {
        this.stream.writeU16(this.checksum.length);
        this.stream.writeBytes(new TextEncoder().encode(this.checksum));
    }
    
    decode(): void {}
}